### DISCUSSION TITLE
Meeting 2024-06-25

### DISCUSSION THREAD

**Discussion by jamesmunns** [rust-embedded/wg#769] [General]
This is the agenda for the meeting today. Please feel free to add items you'd like to discuss below.

**Comment by jamesmunns** [comment]
Short update from Rust Council discussion:

I had a good conversation with some council folks, and I am interested in proposing the creation of a Rust team I'm calling the "Advocacy Team". This is still "pre-pre-RFC" state, but I'm raising the idea with the embedded wg to see how folks feel about it.

The idea here is:

* Have a top level team consisting of "advocacy subteams", for example, "embedded", "user interfaces", "rust in the linux kernel", etc. subteams
* The responsibilities of these subteams would be twofold:
    * To surface the needs/concerns of the domains that they represent, e.g. to be a dedicated person to raise issues, instead of it being ad-hoc
    * To serve as the contact person when the rest of the project has questions, e.g. for RFCs, or addressing issues, OR when the Foundation would like to reach out for questions. e.g. the project pings the embedded advocacy team, who either comments on the item, or takes it to the wider group
* Each subteam would have 1-3 representatives, selected by the groups they represent, serving on some (tbd) basis.

There's some discussion on whether it would be possible to organize the Rust project into three major categories:

* Teams - e.g. "long lived groups inside the project, responsible for some aspect of the project such as code or decision making"
* Projects - e.g. "short lived groups inside the project, responsible for some delegated aspect of the project, such as research or sometimes scoped decision making"
* Ping Groups - e.g. "informal groups interested in a topic, no responsibilities, no authority re code/decisions"

In the long term, there's still discussion of how (working) groups like the embedded-wg relate to the rust project. IMO, one potential (totally not decided!) path would be:

* Have some embedded things move into the rust project, responsible for maintaining crates like `embedded-hal`, or maintaining embedded target definitions. This would be a team/subteam of the project.
* Add a new "embedded advocacy" team, responsible for communicating between the "embedded ecosystem" and "the rust project"
* Potentially make the split between "the rust project" and "the embedded wg" more clear, existing outside the project and represented through the above two subgroups, and maybe continuing on as a more informal "Special Interest Group"

This is all very much up for discussion, and I'd like to hear people's opinions about it before I spend a lot of time writing up an RFC for this!

**Reply by jamesmunns** [reply]
Things discussed:

* Talk about end-game being closing the Launching Pad
* Talk about "Special Interest Groups" outside the Rust Project
* Talk about overlap of Ping Groups and Advocacy teams
* What should/should not become part of the `rust-lang` orga?

**Reply by jamesmunns** [reply]
We probably need to run the "embedded wg as a rust project team" conversation in parallel to this one.

**Comment by rmsyn** [comment]
https://github.com/rust-embedded/riscv/issues/212

**Reply by jamesmunns** [reply]
Hey @rmsyn, any specific aspect to discuss in the meeting? Or just general continuation of the chat from last week?

**Reply by rmsyn** [reply]
Continuation from last week, and there are some additional comments in the issue about the `result` module.

Also, some design discussion about `RangedIndex` types, might be generally applicable to other `rust-embedded` crates.

**Comment by BartMassey** [comment]
I have rewritten the Discovery Book Triage RFC quite a bit and published it as a new PR #770 . Thanks for all the comments on the first version: hopefully this is closer to what people want. Feedback very welcome, as I will start to execute the plan this week if folks are good with it.
